keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resolvers/LaunchConfigurationResolver.kt

```diff
@@ -0,0 +1,80 @@
+package com.netflix.spinnaker.keel.ec2.resolvers
+
+import com.netflix.spinnaker.keel.api.ApiVersion
+import com.netflix.spinnaker.keel.api.Resource
+import com.netflix.spinnaker.keel.api.ec2.ClusterSpec
+import com.netflix.spinnaker.keel.clouddriver.CloudDriverCache
+import com.netflix.spinnaker.keel.ec2.SPINNAKER_EC2_API_V1
+import com.netflix.spinnaker.keel.plugin.Resolver
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+
+@Component
+class LaunchConfigurationResolver(private val cloudDriverCache: CloudDriverCache) : Resolver<ClusterSpec> {
+  override val apiVersion: ApiVersion = SPINNAKER_EC2_API_V1
+  override val supportedKind: String = "cluster"
+  private val log by lazy { LoggerFactory.getLogger(javaClass) }
+
+  override fun invoke(resource: Resource<ClusterSpec>): Resource<ClusterSpec> =
+    resource.run {
```

Comment:
We should return early and short circuit calling `ClusterSpec.withResolvedKeyPairs` if a keypair is already specified in the submitted spec.

Reply:
The challenge here is that there are multiple places where key pairs can appear... Do you suggest checking all places (`defaults` and/or `overrides` for all regions?).